lib/hoteldate.rb

```diff
@@ -0,0 +1,34 @@
+require_relative 'room'
+require_relative 'reservation'
+
+module Hotel
+  class Date
+    
+    attr_reader :id, :occupied
+    
+    def initialize(id)
+      # id is a date object
+      @id = id  
+      # occupied is a hash with key room_id and values of reservation instance
+      @occupied = {}  
+    end
+    
+    # Adds either reservation or hotel blocks to @occupied hash.
+    # Below the parameter is called "reservation" but
+    # logic also works for hotel blocks.
+    def add_occupancy(reservation)
+      @occupied[reservation.room] = reservation
+    end
+    
+    # Returns only reservations and not hotel blocks as per specs.
+    def list_reservations
+      return @occupied.values.select { |value| value.class == Hotel::Reservation }
```

Comment:
Ruby will be looking for names locally first, prioritizing local names, since they're all in the same module, there's no need to namespace `Hotel::Reservation`. `Reservation` should just work! Same with the rest of the namespacing in the code.